crates/ty_server/src/server/api/traits.rs

```diff
@@ -93,10 +93,27 @@ pub(super) trait BackgroundDocumentRequestHandler: RetriableRequestHandler {
 
     fn run_with_snapshot(
         db: &ProjectDatabase,
-        snapshot: DocumentSnapshot,
+        snapshot: &DocumentSnapshot,
         client: &Client,
         params: <<Self as RequestHandler>::RequestType as Request>::Params,
     ) -> super::Result<<<Self as RequestHandler>::RequestType as Request>::Result>;
+
+    fn process(
```

Comment:
Can you please update the documentation for all the traits and the module level documentation as well? I think it'd be useful to specify the difference between "run" and "process"